passes/opt/opt_muxtree.cc

```diff
@@ -291,14 +293,14 @@ struct OptMuxtreeWorker
 		// database of known inactive signals
 		// the payload is a reference counter used to manage the
 		// list. when it is non-zero the signal in known to be inactive
-		vector<int> known_inactive;
+		std::unordered_map<int, int> known_inactive;
```

Comment:
We don't use `std::unordered_map` in Yosys because it introduces cross-platform nondeterminism; use `dict` from hashlib.h for this (it has basically the same interface).

Reply:
I prefer using `std::unordered_map` in Yosys, since the cross-platform non-determinism is limited to iterating over these container. It can also be [more performant](https://github.com/YosysHQ/yosys/pull/4677#issuecomment-2474919965) depending on the use case